draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2014,24 +2023,48 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
 
 <section title="Message Abstraction" anchor="message.abstraction">
 <iref primary="true" item="message abstraction"/>
-<iref item="message"/>
+<iref primary="true" item="message"/>
+<iref primary="true" item="self-descriptive"/>
+<t>
+   Each major version of HTTP defines its own syntax for communicating
+   messages. This section defines an abstract data type for HTTP messages
+   based on a generalization of those message characteristics, common structure,
+   and capacity for conveying semantics. This abstraction is used to define
+   requirements on senders and recipients that are independent of the HTTP
+   version, such that a message in one version can be relayed through other
+   versions without changing its meaning.
+</t>
+<t>
+   A <x:dfn>message</x:dfn> consists of control data to describe and route the
+   message, a headers lookup table of key/value pairs for extending that
+   control data and conveying additional information about the sender, message,
+   payload, or context, a potentially unbounded stream of payload data, and a
+   trailers lookup table of key/value pairs for communicating information
+   obtained while sending the payload.
+</t>
 <t>
-   Each major version of HTTP defines its own syntax for the communication of
-   messages. However, they share a common data abstraction.
+   Framing and control data is sent first, followed by a header section
+   containing fields for the headers table. When a message includes a payload,
+   the payload data is sent after the header section and potentially
+   interleaved with zero or more trailer sections containing fields for the
+   trailers table.
```

Comment:
I think we should explicitly note that trailers are constrained in some protocol versions.

Reply:
We do that in the section on Trailers, to the extent they are constrained. HTTP/2 and 3 do not explicitly constrain trailers, they just don't have a frame to carry them yet.

Reply:
https://github.com/httpwg/http-core/pull/557/commits/3a96888c38b67b28e045f75fc111ac38fb7e5fb7